File: 2966-divide-array-into-arrays-with-max-difference/2966-divide-array-into-arrays-with-max-difference.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[][]}
 */
var divideArray = function(nums, k) {
    nums.sort((a, b) => a - b)
    const res1 = []
    for (let i=0; i < nums.length; i += 3) {
        if (nums[i + 2] - nums[i] <= k) {
            res1.push([nums[i], nums[i + 1], nums[i + 2]])
        } else {
            return []
        }
    }
    return res1
    
    //
    
    nums.sort((a, b) => a - b)
    const res = []
    for (const n of nums) {
        if (!res.length || res.at(-1).length === 3) {
            res.push([n])
        } else {
            if (res.at(-1).some(x => Math.abs(n - x) > k)) {
                return []
            } else {
                res.at(-1).push(n)
            }
        }
    }
    return res
};

File: 2966-divide-array-into-arrays-with-max-difference/2966-divide-array-into-arrays-with-max-difference.py
class Solution:
    def divideArray(self, nums: List[int], k: int) -> List[List[int]]:
        nums.sort()
        res = []
        for i in range(0, len(nums), 3):
            if nums[i + 2] - nums[i] > k:
                return []
            res += [[nums[i], nums[i + 1], nums[i + 2]]]
        return res